coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1441,6 +1441,39 @@ func (s *server) CompleteJob(ctx context.Context, completed *proto.CompletedJob)
 			return nil, xerrors.Errorf("complete job: %w", err)
 		}
 
+		// Insert timings outside transaction since it is metadata.
+		// nolint:exhaustruct // The other fields are set further down.
+		params := database.InsertProvisionerJobTimingsParams{
+			JobID: jobID,
+		}
+		for _, t := range completed.GetWorkspaceBuild().GetTimings() {
+			var start, end time.Time
+			if t.Start != nil {
+				start = t.Start.AsTime()
+			}
+			if t.End != nil {
+				end = t.End.AsTime()
+			}
```

Comment:
I don't believe this is possible right now because in `aggregate()` I prevent returning entries like that, but you're right though I shouldn't rely on _spooky action at a distance._